src/emu/softlist.cpp

```diff
@@ -1133,17 +1118,16 @@ void softlist_parser::parse_data_start(const char *tagname, const char **attribu
 	if (strcmp(tagname, "rom") == 0)
 	{
 		static const char *attrnames[] = { "name", "size", "crc", "sha1", "offset", "value", "status", "loadflag" };
-		const char *attrvalues[ARRAY_LENGTH(attrnames)] = { nullptr };
-		parse_attributes(attributes, ARRAY_LENGTH(attrnames), attrnames, attrvalues);
-
-		const char *name = attrvalues[0];
-		const char *sizestr = attrvalues[1];
-		const char *crc = attrvalues[2];
-		const char *sha1 = attrvalues[3];
-		const char *offsetstr = attrvalues[4];
-		const char *value = attrvalues[5];
-		const char *status = attrvalues[6];
-		const char *loadflag = attrvalues[7];
+		auto attrvalues = parse_attributes(attributes, ARRAY_LENGTH(attrnames), attrnames);
+
+		const char *name = attrvalues[0].c_str();
+		const char *sizestr = attrvalues[1].c_str();
+		const char *crc = attrvalues[2].c_str();
+		const char *sha1 = attrvalues[3].c_str();
+		const char *offsetstr = attrvalues[4].c_str();
+		const char *value = attrvalues[5].c_str();
+		const char *status = attrvalues[6].c_str();
+		const char *loadflag = attrvalues[7].c_str();
```

Comment:
Since you're trying to go to std::string everywhere, why not make these `const std::string &` rather than naked pointers?